### PR TITLE
Handle corner case in Resize op

### PR DESCRIPTION
**Pull request by hariharans29**
**Description**: Handle a corner case in the resize operator where-in an input dim (and the corresponding output dim) could be 0 and proceeding to divide the output dim by the input dim to compute the scale for the dimension results in a NaN.  There are some cases where-in a dimension can be legitimately zero (zero bounding boxes in an object detection model, no keypoints extractable from an image, etc.) and the resize op needs to be able to handle this.

**Motivation and Context**
Noticed while debugging a model run failure.

### CODE REVIEW

File: onnxruntime/core/providers/cpu/tensor/upsample.h [hunk 1]
```diff
@@ -307,7 +307,24 @@ class UpsampleBase {
                                      const std::vector<int64_t>& input_dims,
                                      std::vector<float>& scales) const {
     for (size_t i = 0, end = input_dims.size(); i < end; ++i) {
-      scales[i] = static_cast<float>(output_dims[i]) / static_cast<float>(input_dims[i]);
+      // Handle corner case to avoid dividing by zero in the next step
+      if (input_dims[i] == 0) {
+        // Enforce that output_dim is 0, given that we cannot scale 0 by any factor to
+        // result in any non-zero value
+        ORT_ENFORCE(output_dims[i] == 0,
+                    "Input dim is zero but "
```
Comment (skottmckay) [inline onnxruntime/core/providers/cpu/tensor/upsample.h:315]: Prefer returning Status for things run as part of Compute unless it's a logic error in the code that should never happen. 

nit: does the message need to be spread over so many lines? also mildly helpful to output the value of 'i' so which dimension was bad is known.

File: onnxruntime/core/providers/cpu/tensor/upsample.h [hunk 2]
```diff
@@ -307,7 +307,24 @@ class UpsampleBase {
                                      const std::vector<int64_t>& input_dims,
                                      std::vector<float>& scales) const {
     for (size_t i = 0, end = input_dims.size(); i < end; ++i) {
-      scales[i] = static_cast<float>(output_dims[i]) / static_cast<float>(input_dims[i]);
+      // Handle corner case to avoid dividing by zero in the next step
+      if (input_dims[i] == 0) {
```
Comment (skottmckay) [inline onnxruntime/core/providers/cpu/tensor/upsample.h:311]: Semi-related given this change is about handling a zero. Should Upsample<T>::BaseCompute check if the output size will be 0 after creating Y and exit early?

Comment (hariharans29) [reply onnxruntime/core/providers/cpu/tensor/upsample.h:311]: Added the logic

File: onnxruntime/core/providers/cpu/tensor/upsample.h (hunk 1 above)
Comment (hariharans29) [reply onnxruntime/core/providers/cpu/tensor/upsample.h]: Addressed the nit and adding the dimension to the message. This method doesn't return a status as such. Should we change the signature just to accommodate this ? As such, I think ORT_ENFORCE is not bad in this place. Please let me know if you feel differently.

**Pull request merged**